Makefile

```diff
@@ -0,0 +1,97 @@
+version=3.81
+
+PROJECT_DIR ?= .
+DOCKER_IMAGE_TAG ?= dev
+DOCKER_REGISTRY := ghcr.io
+DOCKER_ORG := darpa-askem
+
+
+
+.SECONDEXPANSION:
+
+.PHONY: all
+all:
+	@echo "make <cmd>"
+	@echo ""
+	@echo "Commands:"
+	@echo "  clean                    - remove bin, dist and node_modules directories"
+	@echo "  images                   - build docker images of all targets"
+	@echo "  image-<target>           - build docker image of a specific target - see below for list of targets"
+	@echo "  publish                  - build and push dock images for deployment"
+	@echo "  publish-<target>         - build and push dock image for a specific target - see below for list of targets"
+	@echo "  lint                     - lint the source code"
+	@echo "  test                     - test the source code"
+	@echo "Targets building (but not pushing) docker images of separately:"
+	@echo "  $(TARGETS)"
+	@echo ""
+	@echo "Variables:"
+	@echo "  DOCKER_IMAGE_TAG=<tag>         - tag to append to each pushed docker image (default: dev)"
+
+
+
+## Targets
+TARGETS :=
+
+TARGETS += hmi-server
+clean-hmi-server:
+	./gradlew :packages:services:hmi-server:clean
+
+image-hmi-server: clean-hmi-server
+	./gradlew :packages:services:hmi-server:build -Dquarkus.container-image.build=true -Dquarkus.container-image.tag=$(DOCKER_IMAGE_TAG)
+
+publish-hmi-server: clean-hmi-server
+	./gradlew :packages:services:hmi-server:build -Dquarkus.container-image.push=true -Dquarkus.container-image.tag=$(DOCKER_IMAGE_TAG)
+
+publish-native-hmi-server: clean-hmi-server
+	./gradlew :packages:services:hmi-server:build -Dquarkus.container-image.push=true -Dquarkus.container-image.tag=$(DOCKER_IMAGE_TAG)-native -Dquarkus.package.type=native
+
+
+
+TARGETS += webapp
+clean-webapp:
+	rm -rf $(PROJECT_DIR)/packages/client/graph-scaffolder/dist
+	rm -rf $(PROJECT_DIR)/packages/client/webapp/dist
+	rm -rf $(PROJECT_DIR)/packages/client/webapp/docker/dist
+
+image-webapp: clean-webapp yarn-install
+	yarn workspace graph-scaffolder tsc --build
+	yarn workspace webapp build
+	mv $(PROJECT_DIR)/packages/client/webapp/dist $(PROJECT_DIR)/packages/client/webapp/docker
+	docker build -t $(DOCKER_REGISTRY)/$(DOCKER_ORG)/webapp:$(DOCKER_IMAGE_TAG) $(PROJECT_DIR)/packages/client/webapp/docker
```

Comment:
lets discuss... ideally would like to move this into the bake file, so that CI can build the images
we should not be building things locally and pushing especially since we are targeting multii platform builds

Reply:
there might be a need for locally building containers to test. I've had to do it numerous time on other projects. I agree we should use buildx and bake files for this. I think it would be good to update this as a new PR as this one is getting large.

Reply:
i'm fine with moving that to a separate PR
and using the bake file you can still build locally but at least it will be more consistent with the CI